RustFxA/FxAWebViewController.swift

```diff
@@ -82,43 +65,29 @@ class FxAWebViewController: UIViewController, WKNavigationDelegate {
     }
 
     override func viewDidLoad() {
-        // If accountMigrationFailed then the app menu has a caution icon, and at this point the user has taken sufficient action to clear the caution.
-        RustFirefoxAccounts.shared.accountMigrationFailed = false
-
         super.viewDidLoad()
         webView.navigationDelegate = self
         view = webView
-
-        RustFirefoxAccounts.shared.accountManager.uponQueue(.main) { accountManager in
-            accountManager.getManageAccountURL(entrypoint: "ios_settings_manage") { [weak self] result in
-                guard let self = self else { return }
-
-                // Handle authentication with either the QR code login flow, email login flow, or settings page flow
-                switch self.pageType {
-                case .emailLoginFlow:
-                    accountManager.beginAuthentication { [weak self] result in
-                        if case .success(let url) = result {
-                            self?.baseURL = url
-                            UnifiedTelemetry.recordEvent(category: .firefoxAccount, method: .emailLogin, object: .accountConnected)
-                            self?.webView.load(URLRequest(url: url))
-                        }
-                    }
-                case let .qrCode(url):
-                    accountManager.beginPairingAuthentication(pairingUrl: url) { [weak self] result in
-                        if case .success(let url) = result {
-                            self?.baseURL = url
-                            UnifiedTelemetry.recordEvent(category: .firefoxAccount, method: .qrPairing, object: .accountConnected)
-                            self?.webView.load(URLRequest(url: url))
-                        }
-                    }
-                case .settingsPage:
-                    if case .success(let url) = result {
-                        self.baseURL = url
-                        self.webView.load(URLRequest(url: url))
-                    }
-                }
+        
+        viewModel.authenticate()
+        
+        viewModel.onEmittingNewState = { [weak self] output in
+            let (request, method) = output
+            
+            if let _method = method {
```

Comment:
It would be better use the same variable name. `_method`

Reply:
Done